lib/dialects/abstract/query-generator.js

```diff
@@ -393,7 +393,7 @@ class QueryGenerator {
 
           suffix += selectFromTmp;
         }
-      } else if (this._dialect.supports.returnValues && options.returning) {
```

Comment:
Keep this check, first check test if dialect supports returnings like MySQL which doesn't support it. Second if user is requesting to return records

Reply:
Look at line 367 - you won't even get into this if statement if `this._dialect.supports.returnValues` was false. why recheck for it?